library/manageiq.py

```diff
@@ -178,13 +187,14 @@ def main():
     username      = module.params['username']
     password      = module.params['password']
     provider_name = module.params['name']
+    provider_type = module.params['type']
     hostname      = module.params['hostname']
     port          = module.params['port']
     token         = module.params['token']
 
     manageiq = ManageIQ(module, url, username, password)
 
-    res_args = manageiq.add_or_update_provider(provider_name, hostname, port, token)
+    res_args = manageiq.add_or_update_provider(provider_name, provider_type, hostname, port, token)
```

Comment:
@cben why would that be useful? `add_or_update_provider` doesn't have default values